src/util/CancellationHandle.cpp

```diff
@@ -23,6 +23,7 @@ void CancellationHandle<WatchDogEnabled>::startWatchDogInternal()
   // Setting the atomic here does synchronize memory for assignment,
   // so this is safe without a mutex
   bool running = watchDogRunning_.exchange(true);
+  // This function is only supposed to be run once.
```

Comment:
If we automatically start the watchdog, then this function can and should be private.